lib/imageflow/graph_runner.ex

```diff
@@ -1,14 +1,36 @@
 defmodule Imageflow.GraphRunner do
-  alias Imageflow.{Graph, Native}
+  alias Imageflow.{Graph, Native, Result}
 
   def run(%Graph{} = graph) do
     with {:ok, job} <- Native.create(),
          :ok <- add_inputs(job, graph.inputs),
          :ok <- add_outputs(job, graph.outputs),
          :ok <- send_task(job, graph),
-         :ok <- save_outputs(job, graph.outputs),
-         :ok <- Native.destroy(job) do
-      :ok
+         :ok <- save_outputs(job, graph.outputs) do
+      {:ok, job, graph}
+    end
+  end
+
+  def get_results(job, %Graph{outputs: outputs} = graph) do
```

Comment:
I know I'm the one who suggested this function in the first place, but the API doesn't look correct. As per Elixir standards, `Graph` should at least be the first argument.
But even with that, the fact that the caller needs to keep track of two separate values (the original `graph`, and the new `job`) makes this a clunky API

Also, coupling this with my other comment about `Native.destroy` being missing, wouldn't it be better to just make this a private function, and call it automatically in the original run flow?